.github/workflows/build.yml

```diff
@@ -12,8 +12,39 @@ on:
 
 # A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
+  rust_binaries:
+    strategy:
+      matrix:
+        os: [windows-latest, macos-latest, ubuntu-latest]
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: 'Checkout source'
+        uses: actions/checkout@28c7f3d2b5162b5ddd3dfd9a45aa55eaf396478b # https://github.com/actions/checkout/commits/v2
+      - name: 'Rust toolchain'
+        uses: actions-rs/toolchain@b2417cde72dcf67f306c0ae8e0828a81bf0b189f # https://github.com/actions-rs/toolchain/commits/v1
+        with:
+          toolchain: stable
+      - name: 'Cache cargo dependencies'
+        uses: actions/cache@d1255ad9362389eac595a9ae406b8e8cb3331f16 # https://github.com/actions/cache/commits/v2
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+            target
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
```

Comment:
Looks good overall, but I'm not sure this is working. I re-ran the jobs just now and some failed. Also, would it need to be keyed by runner.os?

Reply:
Yeah, the caching seems to fail, basically just copied https://github.com/actions/cache/blob/main/examples.md#rust---cargo. 

As far as I understand Cargo's cache includes compiled libraries, so yeah, we'd want to cache those separately for each OS.

Reply:
Yeah my mistake, definitely makes sense to key by os.